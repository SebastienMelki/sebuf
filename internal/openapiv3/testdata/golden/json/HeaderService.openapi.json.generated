{"components":{"schemas":{"FieldViolation":{"description":"FieldViolation describes a single validation error for a specific field.","properties":{"description":{"description":"Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')","type":"string"},"field":{"description":"The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')","type":"string"}},"required":["field","description"],"type":"object"},"HeaderRequest":{"description":"Simple request message","properties":{"data":{"description":"Request data","type":"string"}},"type":"object"},"HeaderResponse":{"description":"Simple response message","properties":{"result":{"description":"Response data","type":"string"}},"type":"object"},"ValidationError":{"description":"ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.","properties":{"violations":{"description":"List of validation violations","items":{"$ref":"#/components/schemas/FieldViolation"},"type":"array"}},"required":["violations"],"type":"object"}}},"info":{"title":"HeaderService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/method-headers":{"post":{"description":"Method with additional method-specific headers","operationId":"WithMethodHeaders","parameters":[{"description":"API authentication key","in":"header","name":"X-API-Key","required":true,"schema":{"example":"123e4567-e89b-12d3-a456-426614174000","format":"uuid","type":"string"}},{"description":"Client version identifier","in":"header","name":"X-Client-Version","required":false,"schema":{"example":"1.2.3","type":"string"}},{"description":"Correlation ID for request tracking","in":"header","name":"X-Correlation-ID","required":false,"schema":{"type":"string"}},{"description":"Unique request identifier for tracing","in":"header","name":"X-Request-ID","required":true,"schema":{"format":"uuid","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"WithMethodHeaders","tags":["HeaderService"]}},"/api/v1/override-header":{"post":{"description":"Method that overrides a service header","operationId":"OverrideServiceHeader","parameters":[{"description":"Override: Special API key for this method","in":"header","name":"X-API-Key","required":true,"schema":{"example":"override-uuid-example","format":"uuid","type":"string"}},{"description":"Client version identifier","in":"header","name":"X-Client-Version","required":false,"schema":{"example":"1.2.3","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"OverrideServiceHeader","tags":["HeaderService"]}},"/api/v1/service-headers":{"post":{"description":"Method with no additional headers (only service headers)","operationId":"ServiceHeadersOnly","parameters":[{"description":"API authentication key","in":"header","name":"X-API-Key","required":true,"schema":{"example":"123e4567-e89b-12d3-a456-426614174000","format":"uuid","type":"string"}},{"description":"Client version identifier","in":"header","name":"X-Client-Version","required":false,"schema":{"example":"1.2.3","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ServiceHeadersOnly","tags":["HeaderService"]}}}}