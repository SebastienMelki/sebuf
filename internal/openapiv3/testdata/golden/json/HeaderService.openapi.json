{"components":{"schemas":{"HeaderRequest":{"description":"Simple request message","properties":{"data":{"description":"Request data","type":"string"}},"type":"object"},"HeaderResponse":{"description":"Simple response message","properties":{"result":{"description":"Response data","type":"string"}},"type":"object"}}},"info":{"title":"HeaderService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/method-headers":{"post":{"description":"Method with additional method-specific headers","operationId":"WithMethodHeaders","parameters":[{"description":"API authentication key","in":"header","name":"X-API-Key","required":true,"schema":{"example":"123e4567-e89b-12d3-a456-426614174000","format":"uuid","type":"string"}},{"description":"Client version identifier","in":"header","name":"X-Client-Version","required":false,"schema":{"example":"1.2.3","type":"string"}},{"description":"Correlation ID for request tracking","in":"header","name":"X-Correlation-ID","required":false,"schema":{"type":"string"}},{"description":"Unique request identifier for tracing","in":"header","name":"X-Request-ID","required":true,"schema":{"format":"uuid","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"WithMethodHeaders","tags":["HeaderService"]}},"/api/v1/override-header":{"post":{"description":"Method that overrides a service header","operationId":"OverrideServiceHeader","parameters":[{"description":"Override: Special API key for this method","in":"header","name":"X-API-Key","required":true,"schema":{"example":"override-uuid-example","format":"uuid","type":"string"}},{"description":"Client version identifier","in":"header","name":"X-Client-Version","required":false,"schema":{"example":"1.2.3","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"OverrideServiceHeader","tags":["HeaderService"]}},"/api/v1/service-headers":{"post":{"description":"Method with no additional headers (only service headers)","operationId":"ServiceHeadersOnly","parameters":[{"description":"API authentication key","in":"header","name":"X-API-Key","required":true,"schema":{"example":"123e4567-e89b-12d3-a456-426614174000","format":"uuid","type":"string"}},{"description":"Client version identifier","in":"header","name":"X-Client-Version","required":false,"schema":{"example":"1.2.3","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/HeaderResponse"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"ServiceHeadersOnly","tags":["HeaderService"]}}}}